dbaas-lib/engines/k8s-psmdb/controller.go

```diff
@@ -243,3 +250,32 @@ func generatePass() ([]byte, error) {
 
 	return b, nil
 }
+
+func (p *PSMDB) checkClusterPods(name string, st *k8sStatus) error {
+	if _, ok := st.Status.Replsets["rs0"]; !ok {
+		return errors.New("no replest `rs0` in cluster status")
+	}
+	for i := 0; i < int(st.Status.Replsets["rs0"].Size); i++ {
+		podData, err := p.cmd.GetObject("pod", name+"-rs0-"+strconv.Itoa(i))
+		if err != nil && !strings.Contains(err.Error(), "Not found") {
+			return errors.Wrap(err, "get pod data")
+		} else if err != nil && strings.Contains(err.Error(), "Not found") {
+			continue
+		}
+		pod := k8s.Pod{}
+		err = json.Unmarshal(podData, &pod)
+		if err != nil {
+			return errors.Wrap(err, "unmarshal pod data")
+		}
+		if pod.Status.Phase != "Pending" {
+			continue
+		}
+		for _, condition := range pod.Status.Conditions {
+			if condition.Status == "False" && strings.Contains(condition.Message, "Insufficient memory") {
+				return errors.New("out of memory")
```

Comment:
Define this error on let's say `dbaas-lib/engines` level. E.g
```go
var ErrOutOfMemory = errors.New("out of memory")
```

So later on you can check 
```go
if err == engines.ErrOutOfMemory ...
...
if errors.Is(err, engines.ErrOutOfMemory)...
```
instead of string comparisons

Reply:
https://github.com/Percona-Lab/percona-dbaas-cli/pull/101/commits/600ae3a23e647cb3f5ddb2e64960c6ca1f41050f